tests/subsys/shell/shell/src/main.c

```diff
@@ -490,9 +491,12 @@ ZTEST(shell, test_section_cmd)
 
 static void *shell_setup(void)
 {
-	/* Let the shell backend initialize. */
-	k_msleep(20);
-	return NULL;
+	const struct shell *sh = shell_backend_dummy_get_ptr();
+
+	/* Wait for the initialization of the shell dummy backend. */
+	while (!shell_ready(sh)) {
+		k_msleep(1);
+	}
```

Comment:
There is `WAIT_FOR` macro for this that can also handle timeout case (should assert on timeout; otherwise, this test will hang during setup without any obvious messages).

Reply:
I have added the missing assert to this code in #49965.